src/backend/distributed/utils/metadata_cache.c

```diff
@@ -886,12 +895,12 @@ BuildDistTableCacheEntry(DistTableCacheEntry *cacheEntry)
 	if (cacheEntry->partitionMethod == DISTRIBUTE_BY_HASH)
 	{
 		TypeCacheEntry *typeEntry = NULL;
-		Node *partitionNode = stringToNode(cacheEntry->partitionKeyString);
-		Var *partitionColumn = (Var *) partitionNode;
 		FmgrInfo *hashFunction = NULL;
 
-		Assert(IsA(partitionNode, Var));
-		typeEntry = lookup_type_cache(partitionColumn->vartype,
+		/* partition column should have been set above */
+		Assert(IsA(cacheEntry->partitionColumn, Var));
+
+		typeEntry = lookup_type_cache(cacheEntry->partitionColumn->vartype,
```

Comment:
I'm not a huge fan of the double-dereference here. I'd stick with assigning to a intermediate `Var *` variable.